### PR TITLE
Create setting for custom CSS URL

### PR DESCRIPTION
**Pull request by ColinMcNeil**
Created a settings field for a custom CSS file. Loaded late enough to override vuetify.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -23,16 +23,23 @@ export default {
     primary() {
       return this.$store.state.settings.theme.primary;
     },
+    custom_css_url(){
+      console.log(this.$store.state.settings.custom_css_url)
+      return this.$store.state.settings.custom_css_url;
+    }
   },
   watch: {
-    primary(hex) {
```
Comment (Alexays) [inline src/App.vue]: Is there a reason you deleted that?

File: src/App.vue [hunk 2]
```diff
@@ -23,16 +23,23 @@ export default {
     primary() {
       return this.$store.state.settings.theme.primary;
     },
+    custom_css_url(){
+      console.log(this.$store.state.settings.custom_css_url)
```
Comment (Alexays) [inline src/App.vue]: Remove debug

File: src/App.vue [hunk 3]
```diff
@@ -23,16 +23,23 @@ export default {
...
-        const { light, secondary } = this.$store.state.settings.theme;
-        this.$vuetify.theme.primary = hex;
-        this.$vuetify.theme.secondary = secondary;
-        this.$vuetify.theme.foreground = light ? '#000000' : '#ffffff';
-      }
-    },
+    custom_css_url(newVal, oldVal){
+      if (newVal === oldVal || oldVal === undefined) return;
+      const url = newVal;
+      console.log(url);
```
Comment (Alexays) [inline src/App.vue]: Remove debug

File: src/components/Settings/index.vue [hunk 4]
```diff
@@ -145,6 +145,15 @@
         <h4 class="headline">Google Analytics</h4>
         <h4 class="subheading">{{ $t('settings.analytics_desc') }}</h4>
         <v-switch :label="$tc('settings.onOff', settings.analytics)" v-model="settings.analytics"/>
+        <h4 class="headline">Custom CSS</h4>
```
Comment (Alexays) [inline src/components/Settings/index.vue]: Need a translation here too

File: src/components/Settings/index.vue [hunk 5]
```diff
@@ -145,6 +145,15 @@
...
+        <h4 class="subheading">{{ $t('settings.custom_css_desc') }}</h4>
+        <v-text-field
+              class="mt-0" hide-details
+              v-model="settings.custom_css_url"
+              :label="URL"/>
```
Comment (Alexays) [inline src/components/Settings/index.vue]: `URL` isn't defined, is it?

File: src/i18n/en.json [hunk 6]
```diff
@@ -55,6 +55,7 @@
     "browse": "Browse",
     "theme": "Theme",
     "analytics_desc": "Just to see how you use cards",
+    "custom_css_desc": "Insert custom CSS URL (https://dapper-cdn.nyc3.digitaloceanspaces.com/test.css)",
```
Comment (Alexays) [inline src/i18n/en.json]: Prefer a generic URL here, e.g. https://exemple.com/style.css

**Comment by ColinMcNeil** [comment]
Should be all of those all set. CSP actually broke when I was testing using [this](https://gist.github.com/ColinMcNeil/49750d5fb04250cb9bd51009020d7a80). The issue is that vuetify doesn't let you change anything except color :/

Also, screenshot of my custom CSS:
image

File: src/components/Settings/index.vue [hunk 7]
```diff
@@ -145,6 +145,12 @@
         <h4 class="headline">Google Analytics</h4>
         <h4 class="subheading">{{ $t('settings.analytics_desc') }}</h4>
         <v-switch :label="$tc('settings.onOff', settings.analytics)" v-model="settings.analytics"/>
+        <h4 class="headline">$t('settings.custom_css')</h4>
```
Comment (Alexays) [inline src/components/Settings/index.vue]: Use interpolation otherwise translation will not work

File: src/components/Settings/index.vue [hunk 8]
```diff
@@ -145,6 +145,12 @@
...
+        <h4 class="subheading">$t('settings.custom_css_desc')</h4>
```
Comment (Alexays) [inline src/components/Settings/index.vue]: here too

**Comment by ColinMcNeil** [comment]
Should be good

**Pull request merged**